forch/device_report_client.py

```diff
@@ -90,13 +90,19 @@ def _convert_and_handle(self, mac, progress):
         return False
 
     def _process_progress(self, mac, session):
+        should_retry = True
         try:
             for progress in session:
+                should_retry = False
                 if self._convert_and_handle(mac, progress):
                     self._logger.warning('Terminal state reached for %s', mac)
             self._logger.error('Progress complete for %s', mac)
         except Exception as e:
             self._logger.error('Progress exception: %s', e)
+        self.disconnect(mac)
+        if should_retry:
```

Comment:
It looks like the session progress can be empty, thus session needs to be restarted. Shouldn't we fix the root cause of why session can be empty coming from DAQ?